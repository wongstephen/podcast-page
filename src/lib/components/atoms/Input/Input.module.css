.container {
  background-color: var(--color-gray-100);
  border: none;

  font-family: var(--font-primary);
  font-size: var(--font-size-md);
  font-weight: var(--font-weight-medium);
  color: var(--color-gray-500);
}
