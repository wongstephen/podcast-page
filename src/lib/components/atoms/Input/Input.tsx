import React from 'react';

import style from './Input.module.css';
import { cn } from '@/lib/utilities/cn';

const Input = ({
  className,
  type = 'text',
  ...props
}: React.InputHTMLAttributes<HTMLInputElement>) => (
  <input type={type} className={cn(style.container, className)} {...props} />
);
Input.displayName = 'Input';

export { Input };
