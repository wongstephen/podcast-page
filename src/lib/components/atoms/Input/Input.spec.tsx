import { render, screen, fireEvent } from '@testing-library/react';
import { Input } from './Input';

describe('Input Component', () => {
  it('renders', () => {
    render(<Input />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Input placeholder="Snapshot Input" />);
    expect(container).toMatchSnapshot();
  });

  it('handles user input correctly', () => {
    const handleChange = jest.fn();
    render(<Input onChange={handleChange} />);
    const inputElement = screen.getByRole('textbox');

    fireEvent.change(inputElement, { target: { value: 'test input' } });
    expect(handleChange).toHaveBeenCalled();
    expect(inputElement).toHaveValue('test input');
  });
});
