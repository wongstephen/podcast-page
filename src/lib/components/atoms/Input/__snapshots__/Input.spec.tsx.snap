// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Input Component matches snapshot 1`] = `
<div>
  <input
    class="container"
    placeholder="Snapshot Input"
    type="text"
  />
</div>
`;
